### PR TITLE
Added configuration for programs service

### PR DESCRIPTION
**Pull request by clintonb**
The non-programs changes focus on cleaning up existing configuration in order to remove duplication and improve readability. The programs-specific configuration has been generated from the django-ida template.

ECOM-2108

@feanil @jibsheet @jimabramson

### CODE REVIEW

File: playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2 [hunk 1]
```diff
@@ -9,7 +9,8 @@
 # and a key being provided via NEWRELIC_LICENSE_KEY
 {{ role_name|upper }}_NEWRELIC_APPNAME: "{{ '{{ COMMON_ENVIRONMENT }}' }}-{{ '{{ COMMON_DEPLOYMENT }}' }}-{{ '{{' }} {{ role_name }}_service_name }}"
 {{ role_name|upper }}_PIP_EXTRA_ARGS: "-i {{ '{{ COMMON_PYPI_MIRROR_URL }}' }}"
-{{ role_name|upper }}_NGINX_PORT: XXX   # TODO Change this!!!
+{{ role_name|upper }}_NGINX_PORT: 80
```
Comment (feanil) [inline playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2]: I liked TODO for this.  Why the change?

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2 [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  ssl_certificate /etc/ssl/certs/{{ NGINX_SSL_CERTIFICATE|basename }};
+  ssl_certificate_key /etc/ssl/private/{{ NGINX_SSL_KEY|basename }};
+  # request the browser to use SSL for all connections
+  add_header Strict-Transport-Security "max-age=31536000; includeSubDomains";
+
+  {% else %}
+  listen {{ PROGRAMS_NGINX_PORT }} {{ default_site }};
+  {% endif %}
+
+  location ~ ^/static/(?P<file>.*) {
```
Comment (feanil) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2:35]: Are these assets hashed?  If so we should add some cache headers so we don't have to always re-serve them.

File: playbooks/roles/programs/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+##
+# Defaults for role programs
+# 
+PROGRAMS_GIT_IDENTITY: !!null
+
+# depends upon Newrelic being enabled via COMMON_ENABLE_NEWRELIC
+# and a key being provided via NEWRELIC_LICENSE_KEY
+PROGRAMS_NEWRELIC_APPNAME: "{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-{{ programs_service_name }}"
+PROGRAMS_PIP_EXTRA_ARGS: "-i {{ COMMON_PYPI_MIRROR_URL }}"
+PROGRAMS_NGINX_PORT: 80
```
Comment (feanil) [inline playbooks/roles/programs/defaults/main.yml]: By default these should be high ports that don't conflict with other apps and don't need root privileges to run.

**Comment by feanil** [comment]
Once port numbers have been fixed, :+1:

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2 (hunk 2 above)
Comment (feanil) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2:35]: talked in person, unsure if these will be hashed initially so leaving it alone for now.

**Comment by jimabramson** [comment]
:+1:

**Comment by jimabramson** [comment]
oh.  :-)

**Comment by clintonb** [comment]
@jimabramson Excellent timing!
